File: flut/.flutter/lib/flutter/widgets/raw_menu_anchor.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flut/error.dart';

class FlutMenuController with FlutRealtimeObject<MenuController> {
  FlutMenuController.createFromData({
    required FlutRuntime runtime,
    required Map<String, dynamic> data,
    required MenuController target,
  }) {
    initRealtimeFromData(runtime: runtime, data: data, target: target);
  }

  FlutMenuController.createFromObject({
    required FlutRuntime runtime,
    required int oid,
    required MenuController target,
  }) {
    initRealtimeFromObject(
      runtime: runtime,
      oid: oid,
      type: 'MenuController',
      target: target,
    );
  }

  static FlutRealtimeObject flutCreate(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return FlutMenuController.createFromData(
      runtime: runtime,
      data: data,
      target: MenuController(),
    );
  }

  @override
  dynamic getRawProperty(String property) {
    switch (property) {
      case 'isOpen':
        return flutTarget.isOpen;
    }
    throw FlutUnknownPropertyException('MenuController', property);
  }

  @override
  bool setProperty(String property, dynamic value) {
    throw FlutUnknownPropertyException('MenuController', property);
  }

  @override
  dynamic callMethod(
    String method,
    List<dynamic> args,
    Map<String, dynamic> kwargs,
  ) {
    switch (method) {
      case 'open':
        final position = kwargs['position'] as Offset?;
        flutTarget.open(position: position);
        return null;
      case 'close':
        flutTarget.close();
        return null;
      case 'closeChildren':
        flutTarget.closeChildren();
        return null;
    }
    throw FlutUnknownMethodException(method);
  }
}

File: flut/flutter/widgets/raw_menu_anchor.py
from flut._flut.engine import FlutRealtimeObject, _flut_pack_value


class MenuController(FlutRealtimeObject):
    _flut_type = "MenuController"

    def __init__(self):
        super().__init__()
        self._flut_create(props={})

    @property
    def isOpen(self) -> bool:
        return bool(self._flut_get("isOpen"))

    def open(self, *, position=None):
        self._flut_call("open", position=_flut_pack_value(position))

    def close(self):
        self._flut_call("close")

    def closeChildren(self):
        self._flut_call("closeChildren")
